src/main/webapp/scripts/map.js

```diff
@@ -154,31 +163,58 @@ function getColourMarker(category) {
 }
 
 /* Place Marker Where Map is Clicked On & Show Popup*/
-function placeMarkerAndPanTo(latLng, map) {
-  var marker = new google.maps.Marker({
+function placeMarkerAndPanTo(latLng) {
+  /* Make marker for the clicked location */
+  const marker = new google.maps.Marker({
     position: latLng,
     map: map,
     icon: greyIcon
   });
-
+  /* Recenter the Map */
   map.panTo(latLng);
-  togglePopup();
+  /* Show the popup box */
+  togglePopup(latLng);
+  /* Update the latitude and longitude values in the popup */
   var markerPosition = marker.getPosition();
   populateLocation(markerPosition);
 
   /* If marker is right-clicked, delete */
-  google.maps.event.addListener(marker, 'rightclick', function(event) {
-      marker.setMap(null);
+  google.maps.event.addListener(marker, "rightclick", function(event) {
+    marker.setMap(null);
   });
 }
 
 /* Set the PopUp to Active */
-function togglePopup() {
+function togglePopup(latLng) {
   document.getElementById("popup-add-recs").classList.toggle("active");
+  document.getElementById("submit-recommendation").addEventListener("click", () =>
+    postMarker(latLng), false);
 }
 
 /* Use the position of marker on map to auto-fill location */
 function populateLocation(pos) {
   var location = pos.lat() + ", " + pos.lng();
   document.getElementById("location").value = location;
 }
+
+/* POST marker */
+function postMarker(latLng) {
+  const params = new URLSearchParams();
+  params.append('lat', latLng.lat());
+  params.append('lng', latLng.lng());
+  fetch('/added-markers', {
+    method: 'POST', 
+    body: params
```

Comment:
Would simple JSON work here? Something like

```
body: {
  "lat": latLng.lat(),
  "lng": latLng.lng()
}
```

I am not sure about the syntax here. The problem is, `URLSearchParams` doesn't seem quite appropriate here as it is meant to be used for params in URL ([ref](http://go/mdn/API/URLSearchParams)). It does give you a nice interface to work with key value pairs but you could simply use JSON format as I mentioned or some generic key value pair type.

Reply:
I'll definitely look into this - I couldn't get this to work, and tried a few things. I'll look further into it and hopefully can get it working.

Reply:
+1, can you please share the error or any relevant information on why it is not working?

Reply:
When I try to submit a recommendation and place a marker, the marker isn't saved  to the datastore and I get this:
![Screenshot 2020-09-10 at 8 49 15 AM - Display 1](https://user-images.githubusercontent.com/43929997/92697108-bea38500-f342-11ea-8e71-4591c45a59bb.png)